### PR TITLE
Closes #133: Updated documentation and test cases for age unit

### LINKED ISSUE #133: Feature Request: Improve handling of age unit in derivations
### Feature Idea

At the moment, the documentation doesn't mention what happens when the age units in the data and the metadata doesn't match. The AI chat bot guesses that the units are converted:

Image

Unfortunately, that isn't true.

I would suggest to describe the behavior of the derivations in the documentation.

The function could also be updated to convert the age in data and metadata to a unified unit ([`derive_var_age_years()`](https://pharmaverse.github.io/admiral/main/reference/derive_var_age_years.html) could be helpful for this). It would also be helpful if the derivations issue a warning if records from the input data couldn't be matched to the metadata and maybe even provide these records (like for duplicates in admiral functions).

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by Lina2689**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiralpeds (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiralpeds site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: vignettes/advs.Rmd [hunk 1]
```diff
@@ -331,15 +384,78 @@ advs <- advs %>%
   )
 ```
 
+#### Age Unit Handling and Standardization {#age-unit-handling}
+
+#### Understanding Age Unit Requirements
+
+The `derive_params_growth_age()` and `derive_params_growth_height()` functions require
```
Comment (bundfussr) [inline vignettes/advs.Rmd]: Is `derive_params_growth_height()` using age? Should it be updated to issue a similar warning for unmatched records?

File: R/derive_params_growth_age.R [hunk 2]
```diff
@@ -122,6 +157,7 @@
 #' @export
 #'
 #' @importFrom dplyr if_else
+#' @importFrom dplyr distinct
 #'
 #' @examples
```
Comment (bundfussr) [inline R/derive_params_growth_age.R:163]: In the example a warning regarding unmatched records is issued. Is this expected?

I had a short look at these 7 records but it was not obvious to me why they didn't match.

File: vignettes/advs.Rmd (hunk 1 above)
Comment (Lina2689) [reply vignettes/advs.Rmd]: derive_params_growth_height() is not using age, updated to remove this.

File: R/derive_params_growth_age.R (hunk 2 above)
Comment (Lina2689) [reply R/derive_params_growth_age.R:163]: Yes, this is expected.

The unmatched records arise due to mismatches in age unit and/or derived age bins used in the join with the metadata, which means no corresponding metadata row exists for these observations. As a result, these records are excluded from the SDS/percentile derivation.

The warning has been updated to clarify this behavior, and the unmatched records are printed to support easier debugging.

Comment (Fanny-Gautier) [inline R/derive_params_growth_age.R:163]: Can we add 2 examples: one with mismatching units (as currently displayed), and one with fully matching units?

File: R/derive_params_growth_age.R [hunk 3]
```diff
@@ -112,6 +130,23 @@
 #'
 #'  If left as default value, `NULL`, then parameter not derived in output dataset.
 #'
+#' @details
```
Comment (Fanny-Gautier) [inline R/derive_params_growth_age.R:133]: Very nice, thank you for reinforcing the importance of matching units!

File: vignettes/advs.Rmd [hunk 4]
```diff
@@ -331,15 +384,77 @@ advs <- advs %>%
   )
 ```
 
+#### Age Unit Handling and Standardization {#age-unit-handling}
+
+#### Understanding Age Unit Requirements
+
+The `derive_params_growth_age()` function require that the age unit in your input data **exactly matches** the age unit in your metadata.
```
Comment (Fanny-Gautier) [inline vignettes/advs.Rmd]:
```suggestion
The `derive_params_growth_age()` function requires that the age unit in your input data **exactly matches** the age unit in your metadata.
```

File: vignettes/advs.Rmd [hunk 5]
```diff
@@ -331,15 +384,77 @@ advs <- advs %>%
...
+  )
+```
+
 Here is how these age variables look:
 
 ```{r, eval=TRUE, echo=FALSE}
 dataset_vignette(
   advs,
-  display_vars = exprs(USUBJID, BRTHDT, ADT, AAGECUR, AAGECURU)
+  display_vars = exprs(USUBJID, BRTHDT, ADT, AAGECUR, AAGECURU, AAGECUR_DAYS, AAGECURU_STD)
```
Comment (Fanny-Gautier) [inline vignettes/advs.Rmd:440]: It looks like the variables are already in days, converted to days? Could it be specified to clarify the above example?

File: vignettes/advs.Rmd
Comment (Fanny-Gautier) [inline vignettes/advs.Rmd]:
```suggestion
* [Metadata](#metadata)
* [Initial ADVS Set-up](#advs_start)
* [Derive Additional Variables for Anthropometric
  indicators](#derive_vars)
  * [Importance of Age Units using
    `derive_params_growth_age()`](#age-unit-handling)    
* [Derive Additional Parameters for Anthropometric
  indicators](#derive_params)
* [Remaining ADVS Set-up](#advs_end)
```
Adding a sub-section to the programming workflow to emphasize the importance of age units.

File: vignettes/advs.Rmd [hunk 6]
```diff
@@ -331,15 +384,77 @@ advs <- advs %>%
   )
 ```
 
+#### Age Unit Handling and Standardization {#age-unit-handling}
+
+#### Understanding Age Unit Requirements
+
+The `derive_params_growth_age()` function require that the age unit in your input data **exactly matches** the age unit in your metadata.
+The functions do **NOT** automatically convert between age units.
```
Comment (Fanny-Gautier) [inline vignettes/advs.Rmd]:
```suggestion
The function does **NOT** automatically convert between age units.
```

File: R/derive_params_growth_age.R [hunk 7]
```diff
@@ -112,6 +130,23 @@
...
+#'
+#' This function requires that the age unit (`age_unit` parameter) in the input dataset
+#' matches the age unit (`AGEU` variable) in the metadata (`meta_criteria`). The function
+#' does NOT perform automatic age unit conversion.
+#'
+#' If records from the input dataset cannot be matched to the metadata due to:
+#' - Mismatched age units (e.g., data in days but metadata in months)
+#' - Missing metadata for the specific age/sex combination
+#'
+#' A warning will be issued listing the number of unmatched records. These records will
```
Comment (Fanny-Gautier) [inline R/derive_params_growth_age.R]:
```suggestion
#' a warning will be issued listing the number of unmatched records. These records will
```

File: tests/testthat/test-derive_params_growth_age.R
Comment (Fanny-Gautier) [inline tests/testthat/test-derive_params_growth_age.R:1]: Maybe use `suppressMessages(library(dplyr))` to avoid the following message:

Image

Could you also please check whether it is expected to have the following ones, and if it can be dropped?
Image

File: DESCRIPTION [hunk 8]
```diff
@@ -56,7 +56,8 @@ Suggests:
     rmarkdown,
     stringr (>= 1.4.0),
     testthat (>= 3.0.0),
-    tibble
+    tibble,
+    here
```
Comment (Fanny-Gautier) [inline DESCRIPTION]: `here` should have already been added within other PRs. Please add it within the alphabetical order if necessary.

**Review by Fanny-Gautier — CHANGES_REQUESTED** [review]
Amazing job Lina!
Clear, precise and very useful, thank you very much for these updates!

File: R/derive_params_growth_age.R (hunk 2 above)
Comment (Fanny-Gautier) [reply R/derive_params_growth_age.R:163]: We can further discuss that comment as necessary.

File: tests/testthat/test-derive_params_growth_age.R
Comment (Lina2689) [reply tests/testthat/test-derive_params_growth_age.R:1]: these are getting displayed as i have added print() to show the unmatched records. so its expected.

File: R/derive_params_growth_age.R [hunk 9]
```diff
@@ -112,6 +130,23 @@
...
+#'
+#' This function requires that the age unit (`age_unit` parameter) in the input dataset
+#' matches the age unit (`AGEU` variable) in the metadata (`meta_criteria`). The function
+#' does NOT perform automatic age unit conversion.
+#'
+#' If records from the input dataset cannot be matched to the metadata due to:
+#' - Mismatched age units (e.g., data in days but metadata in months)
+#' - Missing metadata for the specific age/sex combination
+#'
+#' a warning will be issued listing the number of unmatched records. These records will
```
Comment (Fanny-Gautier) [inline R/derive_params_growth_age.R]: Could we add a precision that the warnings in the example are expected ?
```suggestion
#' a warning (as shown in the example below) will be issued, listing the number of unmatched records. These records will
```

**Review by bundfussr — APPROVED** [review]

**Review by Fanny-Gautier — APPROVED** [review]
LGTM! Thank you for the implementation, this is very helpful!

File: R/derive_params_growth_age.R [hunk 10]
```diff
@@ -142,8 +142,8 @@
 #' - Mismatched age units (e.g., data in days but metadata in months)
 #' - Missing metadata for the specific age/sex combination
 #'
-#' a warning (as shown in the example below) will be issued, listing the number of unmatched records. These records will
-#' NOT appear in the output dataset.
+#' a warning (as shown in the example below) will be issued, listing the number of unmatched records. # nolint
```
Comment (bundfussr) [inline R/derive_params_growth_age.R]: I think `# nolint` should be removed.

File: inst/WORDLIST [hunk 11]
```diff
@@ -31,9 +31,11 @@ funder
 kapa
 measurer
 metadataset
+nolint
```
Comment (bundfussr) [inline inst/WORDLIST]: This shouldn't be added.

File: R/derive_params_growth_age.R (hunk 10 above)
Comment (Lina2689) [reply R/derive_params_growth_age.R]: @bundfussr This line has 101 characters, so its required.

Comment (bundfussr) [reply R/derive_params_growth_age.R]: Why can't you wrap the line, e.g., to 80 characters?

File: inst/WORDLIST (hunk 11 above)
Comment (Fanny-Gautier) [reply inst/WORDLIST]: It does not remove it from the list. Should I remove it manually?

image

Comment (bundfussr) [reply inst/WORDLIST]: If `update_wordlist()` doesn't remove it, it is displayed somewhere in the documentation. You could remove it manually from `WORDLIST` and then call `spelling::spell_check_package()` to find out where "nolint" is displayed.

Comment (Fanny-Gautier) [reply inst/WORDLIST]: nowhere... Thank you for the suggestion!
